src/symbols.h

```diff
@@ -15,14 +15,28 @@ class Symbols {
     static Mutex _parse_lock;
     static bool _have_kernel_symbols;
     static bool _libs_limit_reported;
+    static const void* _main_phdr;
+    static const char* _ld_base;
 
   public:
     static void parseKernelSymbols(CodeCache* cc);
     static void parseLibraries(CodeCacheArray* array, bool kernel_symbols);
 
+    static bool isMainExecutable(const char* image_base, const void* map_end) {
+      return _main_phdr != NULL && _main_phdr >= image_base && _main_phdr < map_end;
+    }
+
+    static bool isLoader(const char* image_base) {
+      return _ld_base != NULL && _ld_base == image_base;
+    }
+
     static bool haveKernelSymbols() {
         return _have_kernel_symbols;
     }
+
+    Symbols();
```

Comment:
Constructor for all-static class?

Reply:
This is to initialize `_ld_base` and `_main_phdr`, I kinda prefer this over mutex+lazy initialization

Reply:
Do we need a mutex here? If not, I'll just initialize them lazily

Reply:
c17d592e63691f44e0fb8cb59fe4a8afb7ec1dd2